File: src/csharp/Intel/Iced/Intel/EncoderInternal/OpCodeInfos.cs
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

#if ENCODER && OPCODE_INFO
using System.Text;

namespace Iced.Intel.EncoderInternal {
	static class OpCodeInfos {
		public static readonly OpCodeInfo[] Infos = CreateInfos();

		static OpCodeInfo[] CreateInfos() {
			var infos = new OpCodeInfo[IcedConstants.CodeEnumCount];
			var encFlags1 = EncoderData.EncFlags1;
			var encFlags2 = EncoderData.EncFlags2;
			var encFlags3 = EncoderData.EncFlags3;
			var opcFlags1 = OpCodeInfoData.OpcFlags1;
			var opcFlags2 = OpCodeInfoData.OpcFlags2;
			var sb = new StringBuilder();
			for (int i = 0; i < infos.Length; i++)
				infos[i] = new OpCodeInfo((Code)i, (EncFlags1)encFlags1[i], (EncFlags2)encFlags2[i], (EncFlags3)encFlags3[i], (OpCodeInfoFlags1)opcFlags1[i], (OpCodeInfoFlags2)opcFlags2[i], sb);
			return infos;
		}
	}
}
#endif

File: src/java/iced-x86/src/main/java/com/github/icedland/iced/x86/info/OpCodeInfos.java
// SPDX-License-Identifier: MIT
// Copyright (C) 2018-present iced project and contributors

package com.github.icedland.iced.x86.info;

import com.github.icedland.iced.x86.internal.IcedConstants;
import com.github.icedland.iced.x86.internal.enc.EncoderData;

/**
 * DO NOT USE: INTERNAL API
 *
 * @deprecated Not part of the public API
 */
@Deprecated
public final class OpCodeInfos {
	/**
	 * DO NOT USE: INTERNAL API
	 *
	 * @deprecated Not part of the public API
	 */
	@Deprecated
	public static final OpCodeInfo[] opCodes = createOpCodes();

	private static OpCodeInfo[] createOpCodes() {
		OpCodeInfo[] infos = new OpCodeInfo[IcedConstants.CODE_ENUM_COUNT];
		int[] encFlags1 = EncoderData.encFlags1;
		int[] encFlags2 = EncoderData.encFlags2;
		int[] encFlags3 = EncoderData.encFlags3;
		int[] opcFlags1 = OpCodeInfoData.opcFlags1;
		int[] opcFlags2 = OpCodeInfoData.opcFlags2;
		StringBuilder sb = new StringBuilder();
		for (int i = 0; i < infos.length; i++)
			infos[i] = new OpCodeInfo(i, encFlags1[i], encFlags2[i], encFlags3[i], opcFlags1[i], opcFlags2[i], sb);
		return infos;
	}
}
